content/posts/2021/reana-0.7.4.md

```diff
@@ -0,0 +1,134 @@
+
+We can set `kubernetes_memory_limit` in every step of workflow specification.
+
+
+```diff
+  ...
+  steps:
+    - name: gendata
+      environment: 'reanahub/reana-env-root6:6.18.04'
+      compute_backend: kubernetes
++     kubernetes_memory_limit: '8Gi'
+      commands:
+      - mkdir -p results && root -b -q 'code/gendata.C(${events},"${data}")'
+    - name: fitdata
+      environment: 'reanahub/reana-env-root6:6.18.04'
+      compute_backend: kubernetes
++     kubernetes_memory_limit: '4Gi'
+      commands:
+      - root -b -q 'code/fitdata.C("${data}","${plot}")'
+```
+
+
+### Custom memory limit for Yadage workflow
+
+We can set `kubernetes_memory_limit` in every stage step under `environment.resources`.
+
+```diff
+  ...
+  stages:
+    - name: gendata
+      dependencies: [init]
+      scheduler:
+        scheduler_type: 'singlestep-stage'
+        parameters:
+          events: {step: init, output: events}
+          gendata: {step: init, output: gendata}
+          outfilename: '{workdir}/data.root'
+        step:
+          process:
+            process_type: 'interpolated-script-cmd'
+            script: root -b -q '{gendata}({events},"{outfilename}")'
+          publisher:
+            publisher_type: 'frompar-pub'
+            outputmap:
+              data: outfilename
+          environment:
+            environment_type: 'docker-encapsulated'
+            image: 'reanahub/reana-env-root6'
+            imagetag: '6.18.04'
+            resources:
+              - compute_backend: kubernetes
++             - kubernetes_memory_limit: '8Gi'
+```
+### Custom memory limit for CWL workflow
+
+We can set `kubernetes_memory_limit` in every step under `hints.reana`.
+
+```diff
+  ...
+  steps:
+    gendata:
+      run: gendata.cwl
+      in:
+        gendata_tool: gendata_tool
+        events: events
+      hints:
+        reana:
+          compute_backend: kubernetes
++         kubernetes_memory_limit: '8Gi'
+      out: [data]
+```
+
+### Custom memory limit administration
+
+REANA cluster administrators can configure the default memory limit for Kubernetes compute backend user job containers by setting
+[kubernetes_jobs_memory_limit](https://github.com/reanahub/reana/tree/master/helm/reana) Helm value and specifying the desired value accordingly.
+
+Also, administrators can set the maximum custom memory limit value that users can assign to their job containers via
+`kubernetes_memory_limit` hint in `reana.yaml` by changing the
+[kubernetes_jobs_max_user_memory_limit](https://github.com/reanahub/reana/tree/master/helm/reana) Helm value.
+
+Please refer to
+[Kubernetes official documentation](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory)
+for expected memory values and note that exceeding these limits will terminate user job containers.
+
+### reana-client improvements
+
+REANA 0.7.4 command line client introduces a wildcard matching patterns support
```

Comment:
REANA 0.7.4 command-line client introduces support for wildcard matching of files for `ls` and `download` commands. This brings useful file name filtering capabilities for workspaces that contain many thousands of files.which might be The release also improved workflow validation of input parameters and environment images for the `validate` command that was introduced previously in [REANA 0.7.3](https://blog.reana.io/posts/2021/reana-0.7.3/).

REANA 0.7.4 fixes several minor issues such as problems with some CWL workflows on HTCondor compute backend due to wrong job command encoding. Please see [full release notes](...) for more information.